### PR TITLE
Fix error message when record to record coercion might be needed

### PR DESCRIPTION
**Pull request by LucGenetier**
Such expression requires record to record coercion which wasn't implemented
Patch(Table2, Lookup(Table2, Table2=GUID("802dcedc-492f-ed11-9db2-0022482aea8f")), {MyDate:  "2022-9-14 7:11:04 pm" })

Before this fix, the error message we were returning was "unknown error", now, it will provide a clear error message, asking for MyDate to be a DateTime and not a string.

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs [hunk 1]
```diff
@@ -483,6 +483,35 @@ public override async ValueTask<FormulaValue> Visit(AggregateCoercionNode node,
...
+                    foreach (var coercion in node.FieldCoercions)
+                    {
+                        CheckCancel();
+                        var newScope = scopeContext.WithScopeValues(record);
+                        var newValue = await coercion.Value.Accept(this, context.NewScope(newScope));
+
+                        if (newValue is ErrorValue)
+                        {
+                            return newValue;
+                        }
```
[the remark this answers is not in the document]
Comment (LucGenetier) [reply src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs]: As discussed, I've removed that code

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Patch.txt [hunk 2]
```diff
@@ -124,4 +149,9 @@ Blank()
 Blank()
 
 >> Patch(t1, {DisplayNameField1:1/0}, {DisplayNameField2:"mars"})
-Blank()
\ No newline at end of file
+Blank()
+
+
+// Record to Record coercion not allowed
+>> Patch(t1, First(Filter(t1, Field2="earth")), {Field3: "2022-11-14 7:22:06 pm"})
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Patch.txt:156]: >>> [](http://example.com/codeflow?start=0&length=2)

this test is the key thing we're enabling....

**Comment by MikeStall** [comment]
>         public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)

these changes are all coming from PowerApps?

---
Refers to: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/PatchFunction.cs:185 in e31f286. [](commit_id = e31f286cd071bea748908eb5031951627eb4e9ad, deletion_comment = False)

**Comment by LucGenetier** [comment]
>         public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)

Absolutely

---
In reply to: [1248441649](https://github.com/microsoft/Power-Fx/pull/666#issuecomment-1248441649) [](http://example.com/codeflow?ancestors=1248441649,1248441649)

---
Refers to: src/libraries/Microsoft.PowerFx.Interpreter/Functions/Mutation/PatchFunction.cs:185 in e31f286. [](commit_id = e31f286cd071bea748908eb5031951627eb4e9ad, deletion_comment = False)

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**